### DISCUSSION TITLE
HTTP module

### DISCUSSION THREAD

**Discussion by m-henderson** [odin-lang/Odin#972] [Ideas/Requests]
**Is your feature request related to a problem? Please describe.**
is there a way yet to create a simple webserver with odin? I am wanting to contribute to the [Hucelia ](https://github.com/hucelia/hucelia)("web framework for odin") but the team is just waiting on there to be a way to create a simple webserver with Odin. Like an HTTP module in Odin or something. 

**Describe the solution you'd like**
I need a way to create a simple web server that can serve files via http request. 

**Additional context**
let me know if you need more context from me

**Comment by gingerBill** [comment]
Networking is planned for the core library, this includes http.

**Comment by m-henderson** [comment]
@gingerBill is it already in the process? I would like to help.

**Comment by dotbmp** [comment]
@m-henderson I've started binding Winsock2 with the intent of building HTTP support on top

There is already a Berkeley sockets binding. See the link at https://github.com/odin-lang/odin-libs

I plan to take Winsock and the Berkeley sockets and make a platform-agnostic socket library, then build out HTTP support on top. I also left some more details as an issue on the Hucelia repo. If there's a specific way you'd like to help out in this let me know here or on the Odin Discord server.

**Comment by m-henderson** [comment]
@bpunsky thanks pal!

**Comment by hasenj** [comment]
> but the team is just waiting on there to be a way to create a simple webserver with Odin.

I'd like to point out that if you just keep waiting for others to do something before you take the next step you might never get anywhere.

For what it's worth I'm trying to experiment with something in this space, but it's not likely that I will produce anything useful for a long time since it's just something I do when I have free time.

**Comment by gingerBill** [comment]
The eventual socket library `package net` will need to be written in pure Odin and cannot link against third-party libraries.

I am currently planning the design of the Core Library in this project: https://github.com/odin-lang/Odin/projects/2

**Comment by m-henderson** [comment]
@hasenj good point! I have begin working on something. 

@gingerBill when you say core library are you referring to the http library that will be within Odin?

**Comment by gingerBill** [comment]
@m-henderson Yes. The core library is the library that comes with the compiler. Some people call it the _standard library_ but I call it the _core library_ (for a few years).

**Comment by m-henderson** [comment]
@gingerBill what is an issue number that I can help with that is not too complicated? I am new to development on compilers.

**Comment by hasenj** [comment]
> The eventual socket library package net will need to be written in pure Odin and cannot link against third-party libraries.

Do you mean it won't even use the c socket api under the hood?

**Comment by gingerBill** [comment]
@hasenj It will use the OS-specific libraries rather than linking to third-party libraries.

**Comment by renlite** [comment]
I would like to know when to expect the (core) HTTP Module?

There exists a similar module https://github.com/laytan/odin-http.

**Comment by MrBns** [comment]
its 2025. what is the planning for core http module ?

**Comment by renlite** [comment]
Is there a roadmap 2025 for Odin?

**Comment by JustForCodin** [comment]
2025 ends pretty soon. Any updates? :D

**Reply by MrBns** [reply]
this language maybe not designed to use for public use case. they don't care anything. they just care about their sponsor.
